gst/gst_element_factory.go

```diff
@@ -28,7 +28,48 @@ func NewElementWithName(factory string, name string) (*Element, error) {
 		elem = C.gst_element_factory_make((*C.gchar)(elemName), (*C.gchar)(cname))
 	}
 	if elem == nil {
-		return nil, fmt.Errorf("Could not create element: %s", factory)
+		return nil, fmt.Errorf("could not create element: %s", factory)
+	}
+	return wrapElement(glib.TransferNone(unsafe.Pointer(elem))), nil
+}
+
+// Create a new element of the type defined by the given elementfactory. The supplied list of properties, will be passed at object construction.
+//
+// using this results in a free() panic :( DO NOT USE
+func NewElementWithProperties(factory string, properties map[string]interface{}) (*Element, error) {
+	props := make([]*C.char, 0)
+	values := make([]C.GValue, 0)
+
+	for p, v := range properties {
+		cpropName := C.CString(p)
+		defer C.free(unsafe.Pointer(cpropName))
+
+		props = append(props, cpropName)
+
+		cValue, err := glib.GValue(v)
+
+		if err != nil {
+			return nil, err
+		}
+
+		values = append(values, *(*C.GValue)(cValue.Unsafe()))
+	}
+
+	cfactory := C.CString(factory)
```

Comment:
Ah...so is this the `free()` error you were talking about. This is an interesting case, because the function wants a constant string. Leading me to believe that if it were to disappear under it that could cause problems.

Reply:
I will lean on ChatGPT a bit:

> const gchar * factoryname: This parameter specifies the name of the factory that creates the element. It is a string (const gchar pointer), which means that the function only needs to read the value of this string and does not take ownership of the memory for the string. Therefore, the caller is responsible for managing the memory of factoryname.

The "not taking ownership" part was confirmed by slomo on #gstreamer, so the `free()` is necessary here

Reply:
I wouldn't expect gstreamer to take ownership of it. But it might try to reference it later, assuming it's a constant. But I'll lean on what slomo thinks.

Reply:
I think I tried even not freeing any of the strings, I always either got a "double free()" panic or some different free panic (cannot remember right now)